src/server.ts

```diff
@@ -80,6 +91,32 @@ const build = (opts: FastifyServerOptions): FastifyInstance => {
       return;
     }
 
+    const authHeader = req.headers.authorization as string;
+    if (authHeader && authHeader.startsWith('Basic ')) {
+      if (!req.routeOptions.url?.startsWith('/api')) {
+        reply.status(404).send({ error: 'not found' });
+        return;
+      }
+
+      const { domain } = req.query as { [key: string]: string };
```

Comment:
is basic auth really what we want to use? why not just have an authenticated API? imo we should require users of the API to use `/login` (one way to authenticate to UMT), get issued a cookie, and then include it in requests to `/api`.  that seems straightforward and not terribly burdensome to users of the API (atm just jonathan)